src/libs/sensors-logging.ts

```diff
@@ -427,29 +427,51 @@ class DataLogger {
     const keysLogPointsInRange = availableLogsKeys.filter((key) => {
       const epochString = Number(key.split('=')[1])
       const logPointDate = new Date(epochString)
-      return logPointDate >= initialTime && logPointDate <= finalTime
+      // Consider logs that start a little before the initial time and end a little after the final time, so we don't
+      // miss data on the start and end of the file.
+      const aLittleBitBeforeInitialTime = new Date(initialTime.getTime() - 10000)
+      const aLittleBitAfterFinalTime = new Date(finalTime.getTime() + 10000)
+      return logPointDate >= aLittleBitBeforeInitialTime && logPointDate <= aLittleBitAfterFinalTime
+    })
+
+    // Generate a object with the initial and final epoch of each log point
+    // To know the initial epoch, check the name of the key. To know the final epoch, check the epoch of the next point.
+    const infoLogPointsInRange = keysLogPointsInRange.map((key, index, array) => {
+      const epochString = Number(key.split('=')[1])
+      const initialDate = new Date(epochString)
+
+      let finalDate: Date | null = null
+      if (index !== array.length - 1) {
+        const nextEpochString = Number(array[index + 1].split('=')[1])
+        finalDate = new Date(nextEpochString)
+      }
+
+      return { key, initialDate, finalDate }
     })
 
     if (keysLogPointsInRange.length === 0) {
       throw new Error('No log points found in the given range.')
     }
 
     const logPointsInRange: CockpitStandardLogPoint[] = []
-    for (const key of keysLogPointsInRange) {
-      const log = (await this.cockpitTemporaryLogsDB.getItem(key)) as CockpitStandardLogPoint
+    for (const info of infoLogPointsInRange) {
+      const log = (await this.cockpitTemporaryLogsDB.getItem(info.key)) as CockpitStandardLogPoint
 
       // Only consider real log points(objects with an epoch and data property, and non-empty data)
       if (log.epoch === undefined || log.data === undefined || Object.keys(log.data).length === 0) continue
 
+      // Exclude those log points that end before the initial time or start after the final time
+      if ((info.finalDate && info.finalDate < initialTime) || info.initialDate > finalTime) continue
+
       logPointsInRange.push(log)
     }
 
     // Sort the log points by epoch, generate a final log file and put in in the local database
     const sortedLogPoints = logPointsInRange.sort((a, b) => a.epoch - b.epoch)
-    const finalLog = sortedLogPoints.map((logPoint) => ({
-      ...logPoint,
-      ...{ seconds: differenceInSeconds(new Date(logPoint.epoch), initialTime) },
-    }))
+    const finalLog = sortedLogPoints.map((logPoint) => {
+      const seconds = Math.max(0, differenceInSeconds(new Date(logPoint.epoch), initialTime))
```

Comment:
There should probably be a comment here explaining that the `max` clips the first point to the start of the log - took me a while to confirm the logic.